lib/tasks/learn_to_rank.rake

```diff
@@ -41,48 +41,56 @@ namespace :learn_to_rank do
     end
   end
 
-  namespace :reranker do
-    desc "Train a reranker model with relevancy judgements"
-    task :train, [:svm_dir, :model_dir] do |_, args|
-      assert_ltr!
-
-      model_dir = args.model_dir || "tmp/libsvm"
-      svm_dir = args.svm_dir || "tmp/ltr_data"
-      sh "env OUTPUT_DIR=#{model_dir} TRAIN=#{svm_dir}/train.txt VALI=#{svm_dir}/validate.txt TEST=#{svm_dir}/test.txt ./ltr_scripts/train.sh"
+  desc "Pull learn to rank model from S3"
+  task :pull_model, [:model_filename] do |_, args|
+    bucket_name = ENV["AWS_S3_RELEVANCY_BUCKET_NAME"]
+    raise "Missing required AWS_S3_RELEVANCY_BUCKET_NAME" if bucket_name.blank?
+
+    models_dir = ENV["TENSORFLOW_MODELS_DIRECTORY"]
+    raise "Please specify the Tensorflow models directory" if models_dir.blank?
+
+    prefix            = "ltr"
+    s3_objects        = Aws::S3::Bucket.new(bucket_name).objects(prefix: prefix)
+    model_files       = s3_objects.map { |object| object.key.delete("#{prefix}/") }
+    latest_model      = model_files.max_by(&:to_i)
+    model_filename    = args.model_filename || latest_model
```

Comment:
This isn't introduced by your change, but might be nice to skip the lookup of the latest_model if `args.model_filename` is already present.

Something like

```ruby
args.model_filename || begin
    # fetch latest_model
end
```